### DISCUSSION TITLE
How could I manually parse the sections of binary?

### DISCUSSION THREAD

**Discussion by trcrsired** [NationalSecurityAgency/ghidra#5557] [Q&A]

**Comment by ryanmkurtz** [comment]
Can you please provide more information?

**Comment by trcrsired** [comment]
> Can you please provide more information?

for example, how to get an import section in a PE binary?

**Comment by trcrsired** [comment]
PEResource.java:8: error: package generic.continues does not exist
import generic.continues.*;

**Comment by ryanmkurtz** [comment]
This should get you started:

```java
ByteProvider provider = new MemoryByteProvider(currentProgram.getMemory(), currentProgram.getImageBase());
PortableExecutable pe = new PortableExecutable(provider, SectionLayout.MEMORY, true, false);
```

From the `pe` object, you can get the rest of the header pieces.
